README.md

```diff
@@ -1,100 +1,275 @@
+
+Use good TDD practices, and test _before_ you code. Remember: red-green-refactor.
+
+## Wave 2: Building the API
+In this wave, you will implement the API described below. The endpoints are described more-or-less in order of complexity, and we recommend you build them in that order. Every endpoint must serve JSON data, and must use HTTP response codes to indicate the status of the request.
+
+Because this API will be used as the backend for a future project, there are strict requirements about how it should be structured. To this end, we have provided a simple script that uses HTTParty to exercise all the endpoints. **This script is not a substitute for writing your own tests!!!!!** There are many edge and negative cases it doesn't cover - these are your responsibility.
+
+The schema of your database and the structure of your rails app are completely up to you, so long as the API conforms to the description and provided script.
+
+### Query Parameters
+Any endpoint that returns a list should accept 3 _optional_ query parameters:
+
+| Name   | Value   | Description
+|--------|---------|------------
+| `sort` | string  | Sort objects by this field
+| `n`    | integer | Number of responses to return per page
+| `p`    | integer | Page of responses to return
+
+So, for an API endpoint like `GET /customers`, the following requests should be valid:
+- `GET /customers`: All customers, sorted by ID
+- `GET /customers?sort=name`: All customers, sorted by name
+- `GET /customers?n=10&p=2`: Customers 10-19, sorted by ID
+- `GET /customers?sort=name&n=10&p=2`: Customers 10-19, sorted by name
+
+Things to note:
+- Sorting by ID is the rails default
+- Possible values for `sort` will be specified
+- If the client requests both sorting and pagination, pagination should be relative to the sorted order
+- Check out the [will_paginate gem](https://github.com/mislav/will_paginate)
+
+### Error Handling
+If something goes wrong, your API should return an appropriate [HTTP status code](http://billpatrianakos.me/blog/2013/10/13/list-of-rails-status-code-symbols/), as well as a list of errors. The list should be formatted like this:
+
+```json
+{
+  "errors": {
+    "sort": "Invalid sort field 'gnome'"
+  }
+}
+```
+
+All errors your API can return should be covered by at least one test case.
+
+### Testing
+Because APIs are often open to the public, thorough testing is essential. For a Rails API, that means controller testing.
+
+For each API endpoint, you should have _at least_:
+- A basic test with no parameters, if applicable
+- Positive and negative tests for each query parameter (`sort`, `n`, `p`)
+- Positive and negative tests for any URI parameters (user ID, movie name)
+- Testing around any data in the request body
+
+Use good TDD practices, and test _before_ you code. Remember: red-green-refactor.
+
+## API Description
```

Comment:
Overall, I think that this is too many requirements for a four day project so I think we need to scale down